music21/meter/tests.py

```diff
@@ -47,7 +47,7 @@ def testCompound(self):
         meterStrDenominator = [1, 2, 4, 8, 16, 32]
         meterStrNumerator = [2, 3, 4, 5, 6, 7, 9, 11, 12, 13]
 
-        for i in range(30):
+        for i in range(8):
             msg = []
             for j in range(1, random.choice([2, 4])):
                 msg.append('%s/%s' % (random.choice(meterStrNumerator),
```

Comment:
Up to 20 of the 40 seconds are taken by this test, even after I reduced the # of iterations. (sometimes takes as little as 3 seconds). TimeSignature creation is the chokepoint. Takes 4.5 seconds to create `meter.TimeSignature('9/4+6/8+13/1')`.

Could be moved to TestSlow.

Reply:
This diff gets the creation time down to 3s from 4.2ish, by saving 261,120 opFrac calls, but not sure if you want to change the signature of `offsetToDepth()` over it.

len(meterSequence.flat) = 512
512 calls to offsetToDepth and offsetToIndex
for each call in offsetToIndex, 1 - 512 loop iterations, all but one with 2 calls to opFrac

261120 calls to opFrac saved

```diff
diff --git a/music21/meter/core.py b/music21/meter/core.py
index 6aee8858d..7382c5ed9 100644
--- a/music21/meter/core.py
+++ b/music21/meter/core.py
@@ -15,6 +15,7 @@ This module defines two component objects for defining nested metrical structure
 :class:`~music21.meter.core.MeterTerminal` and :class:`~music21.meter.core.MeterSequence`.
 '''
 import copy
+from typing import Optional
 
 from music21 import prebase
 from music21.common.numberTools import opFrac
@@ -1765,7 +1766,7 @@ class MeterSequence(MeterTerminal):
         iMatch = self.offsetToIndex(qLenPos)
         return opFrac(self[iMatch].weight)
 
-    def offsetToDepth(self, qLenPos, align='quantize'):
+    def offsetToDepth(self, qLenPos, align='quantize', index: Optional[int] = None):
         '''
         Given a qLenPos, return the maximum available depth at this position
 
@@ -1787,6 +1788,8 @@ class MeterSequence(MeterTerminal):
         >>> b.offsetToDepth(-1)
         Traceback (most recent call last):
         music21.exceptions21.MeterException: cannot access from qLenPos -1.0
+
+        `index` can be provided, if known, for some long MeterSequences to improve performance.
         '''
         qLenPos = opFrac(qLenPos)
         if qLenPos >= self.duration.quarterLength or qLenPos < 0:
@@ -1797,7 +1800,9 @@ class MeterSequence(MeterTerminal):
         # need to quantize by lowest level
         mapMin = self.getLevelSpan(self.depth - 1)
         msMin = self.getLevel(self.depth - 1)
-        qStart, unused_qEnd = mapMin[msMin.offsetToIndex(qLenPos)]
+        if index is None:
+            index = msMin.offsetToIndex(qLenPos)
+        qStart, unused_qEnd = mapMin[index]
         if align == 'quantize':
             posMatch = opFrac(qStart)
```

Reply:
That is fine -- this is a deep dive into the internals, and works fine if index isn't used, so that's great!